### PR TITLE
Feat/permissions support

### PR DESCRIPTION
**Pull request by monitz87**

### CODE REVIEW

File: src/PolymathBase.ts [hunk 1]
```diff
@@ -561,4 +654,45 @@ export class PolymathBase extends PolymathAPI {
...
+    } else if (role === SecurityTokenRole.PermissionsAdministrator) {
+      moduleName = ModuleName.GeneralPermissionManager;
+      permission = Perm.Admin;
+    } else if (
+      [
+        SecurityTokenRole.Erc20DividendsAdministrator,
+        SecurityTokenRole.Erc20DividendsOperator,
+      ].includes(role)
+    ) {
+      moduleName = ModuleName.GeneralTransferManager;
```
Comment (VictorVicente) [inline src/PolymathBase.ts]: Shouldn't this one be `ModuleName.ERC20DividendCheckpoint `?

File: src/PolymathBase.ts [hunk 2]
```diff
@@ -561,4 +654,45 @@ export class PolymathBase extends PolymathAPI {
...
+      permission =
+        role === SecurityTokenRole.Erc20DividendsAdministrator ? Perm.Admin : Perm.Operator;
+    } else if (
+      [
+        SecurityTokenRole.EtherDividendsAdministrator,
+        SecurityTokenRole.EtherDividendsOperator,
+      ].includes(role)
+    ) {
+      moduleName = ModuleName.ERC20DividendCheckpoint;
```
Comment (VictorVicente) [inline src/PolymathBase.ts]: Shouldn't this one be `ModuleName.EtherDividendCheckpoint`?

File: src/types/index.ts [hunk 3]
```diff
@@ -375,7 +400,7 @@ export interface ProcedureArguments {
   [ProcedureType.PauseSto]: PauseStoProcedureArgs;
   [ProcedureType.ControllerTransfer]: ControllerTransferProcedureArgs;
   [ProcedureType.SetController]: SetControllerProcedureArgs;
-  [ProcedureType.ChangeDelegatePermission]: ChangeDelegatePermissionProcedureArgs;
+  [ProcedureType.AssignSecurityTokenRole]: AssignSecurityTokenRoleProcedureArgs;
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/types/index.ts]: Oversight

File: src/types/index.ts [hunk 4]
```diff
@@ -436,8 +456,45 @@ export function isPojo(pojo: any): pojo is Pojo {
...
+  EtherDividendsAdministrator = 'EtherDividendsAdministrator',
+  ShareholdersAdministrator = 'ShareholdersAdministrator',
+}
+
+export enum StoRole {
+  StoOperator = 'StoOperator',
+  StoAdministrator = 'StoAdministrator',
+}
+
+export type Module =
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/types/index.ts]: Would be ideal, yes

File: src/procedures/AssignSecurityTokenRole.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    const delegate = conversionUtils.checksumAddress(delegateAddress);
+
+    try {
+      await contractWrappers.tokenFactory.getSecurityTokenInstanceFromTicker(symbol);
+    } catch (err) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `There is no Security Token with symbol ${symbol}`,
+      });
+    }
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/procedures/AssignSecurityTokenRole.ts]: Makes sense

File: src/PolymathBase.ts (hunk 1 above)
Comment (monitz87) [reply src/PolymathBase.ts]: Yeah, my bad

File: src/PolymathBase.ts (hunk 2 above)
Comment (monitz87) [reply src/PolymathBase.ts]: Yeah, my bad

File: src/procedures/AssignSecurityTokenRole.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      { moduleName: ModuleName.GeneralPermissionManager, symbol },
+      { unarchived: true }
+    ))[0];
+
+    if (!permissionModule) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: 'You must enable the Permissions feature',
+      });
+    }
```
[the remark this answers is not in the document]
Comment (monitz87) [reply src/procedures/AssignSecurityTokenRole.ts]: Makes sense

File: src/types/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (VictorVicente) [reply src/types/index.ts]: I'll create a task to make it happen

**Pull request merged**